packages/firestore/src/core/query.ts

```diff
@@ -540,6 +550,28 @@ export class RelationFilter extends Filter {
         other.internalValue.find(element => element.isEqual(this.value)) !==
           undefined
       );
+    } else if (this.op === RelationOp.IN) {
+      if (this.value instanceof ArrayValue) {
+        return (
+          this.value.internalValue.find(element => element.isEqual(other)) !==
+          undefined
+        );
+      } else {
+        fail('IN filter has invalid value: ' + this.value.toString());
+        return false;
```

Comment:
You can actually `return fail(...)` here rather than have a misleading `return false;` that'll never get hit.